docs/TESTCASE_DEV_GUIDE.md

```diff
@@ -21,64 +22,78 @@
+          vars:
+            skip_msg: "Skip test case '{{ ansible_play_name }}' due to guest OS not support this test case."
+            skip_reason: "Not Supported"
+          when: guest_version == "version_1"
+
+        - name: "Skip test case"
+          include_tasks: ../../common/skip_test_case.yml
+          vars:
+            skip_msg: "Skip test case '{{ ansible_play_name }}' due to vCenter info is not provided."
+            skip_reason: "Blocked"
+          when: vc_info == ""
+
+        - name: "Run testing"
+          block:
+            - name: "Do configurations and get results"
+              include_tasks: new_test_case_task.yml
             
-            - name: Verify task result
-              assert:
+            - name: "Verify results"
+              ansible.builtin.assert:
                 that:
                   - task_result == expected_value
-                fail_msg: "test_result must be expected_value"
-                success_msg: "test_result is expected_value"
-
-            # set test result to 'Passed' when check point verified
-            - include_tasks: ../../common/print_test_result.yml
-              vars:
-                test_result: "Passed"
+                fail_msg: "{{ test_result }} is not the expected one: {{ expected_value }}"
+                success_msg: "{{ test_result }} is as expected: {{ expected_value }}"
 
-          when: vmtools_is_installed
+          when: vm_config == "config_2"
       rescue:
-        - include_tasks: ../setup/test_rescue.yml
+        - name: "Test case failure"
+          include_tasks: ../../common/test_rescue.yml
 ```
 
 #### Test case "block"
-1. In the "block" part, the first included task is "linux/setup/test_setup.yml", this task file is executed at the beginning of each test case. It will fulfill below functions mainly:
-* check base snapshot existence status, revert to it if exists or take base snapshot if does not exist,
-* get VMware Tools status,
+1. In the "block" part, the first included task is "setup/test_setup.yml", will fulfill below functions mainly:
+* check base snapshot existence status, revert to it if it exists,
+* if base snapshot does not exist, do required configurations in guest OS and take a base snapshot,
+* get VMware Tools info,
 * get VM and guest OS info,
-* get guest IP address and add to inventory.
+* get guest OS IP address and add to inventory.
 
 2. In the "block" part, write the tasks of test case execution and verification.
 
-3. In the "block" part, use this task "common/print_test_result.yml" to set test case result as below:
-* "No Run": test case requirements or dependency not meet, 
-* "Failed": test case verification failed,
-* "Passed": test case passed.
+3. In the "block" part, use this task "common/skip_test_case.yml" to set test case result as below:
+* Blocked: Test case dependency is not meet, e.g. no VMtools or no VC for GOSC testing
+* Not Supported: Tested function is not supported by the OS.
```

Comment:
Test case could be not supported by either the guest OS or the VM's config (e.g. guest id, hardware version).